datanode/gateway/middleware.go

```diff
@@ -193,34 +192,31 @@ func (s *SubscriptionRateLimiter) WithSubscriptionRateLimiter(next http.Handler)
 	})
 }
 
-func (s *SubscriptionRateLimiter) WithGrpcInterceptor() grpc.UnaryServerInterceptor {
-	return func(ctx context.Context,
-		req interface{},
-		_ *grpc.UnaryServerInfo,
-		handler grpc.UnaryHandler,
-	) (resp interface{}, err error) {
-		// Fish out IP address from context
-		addr, ok := contextutil.RemoteIPAddrFromContext(ctx)
-		if !ok {
+type ipGetter func(ctx context.Context, method string, log *logging.Logger) string
+
+func (s *SubscriptionRateLimiter) WithGrpcInterceptor(ipGetterFunc ipGetter) grpc.StreamServerInterceptor {
+	return func(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
```

Comment:
non-blocking: `any` looks cleaner (to my eye at least) than `interface{}`

Reply:
yes fair enough, that was copy-pasted from the definition inside grpc :)